Microsoft.Toolkit.Uwp.UI.Controls/TokenizingTextBox/TokenizingTextBox.cs

```diff
@@ -0,0 +1,228 @@
+﻿using System;
+using System.Collections.Generic;
+using Windows.System;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// A text input control that auto-suggests and displays token items.
+    /// </summary>
+    [TemplatePart(Name = PART_AutoSuggestBox, Type = typeof(AutoSuggestBox))]
+    [TemplatePart(Name = PART_WrapPanel, Type = typeof(WrapPanel))]
+    public partial class TokenizingTextBox : Control
+    {
+        private const string PART_AutoSuggestBox = "PART_AutoSuggestBox";
+        private const string PART_WrapPanel = "PART_WrapPanel";
+
+        private AutoSuggestBox _autoSuggestBox;
+        private WrapPanel _wrapPanel;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TokenizingTextBox"/> class.
+        /// </summary>
+        public TokenizingTextBox()
+        {
+            DefaultStyleKey = typeof(TokenizingTextBox);
+        }
+
+        /// <inheritdoc/>
+        protected override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+
+            if (_autoSuggestBox != null)
+            {
+                _autoSuggestBox.QuerySubmitted -= AutoSuggestBox_QuerySubmitted;
+                _autoSuggestBox.SuggestionChosen -= AutoSuggestBox_SuggestionChosen;
+                _autoSuggestBox.TextChanged -= AutoSuggestBox_TextChanged;
+                _autoSuggestBox.KeyDown -= AutoSuggestBox_KeyDown;
+                _autoSuggestBox.CharacterReceived -= AutoSuggestBox_CharacterReceived;
+            }
+
+            _autoSuggestBox = (AutoSuggestBox)GetTemplateChild(PART_AutoSuggestBox);
+            _wrapPanel = (WrapPanel)GetTemplateChild(PART_WrapPanel);
+
+            if (_autoSuggestBox != null)
+            {
+                _autoSuggestBox.QuerySubmitted += AutoSuggestBox_QuerySubmitted;
+                _autoSuggestBox.SuggestionChosen += AutoSuggestBox_SuggestionChosen;
+                _autoSuggestBox.TextChanged += AutoSuggestBox_TextChanged;
+                _autoSuggestBox.KeyDown += AutoSuggestBox_KeyDown;
+                _autoSuggestBox.CharacterReceived += AutoSuggestBox_CharacterReceived;
+            }
+
+            var selectAllMenuItem = new MenuFlyoutItem
+            {
+                Text = "Select All"
```

Comment:
We should localize this, reference example here: https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/e62a883fdacc6da171fc4f74dfa97522898274f7/Microsoft.Toolkit.Uwp.UI.Controls/BladeView/BladeItem.cs#L71

or
"Path".GetLocalized("Key")

https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/master/Microsoft.Toolkit.Uwp.UI.Controls/Strings/en-US/Resources.resw